### DISCUSSION TITLE
AIO v7.9.1 (Nextcloud 27.1.5) + LDAP + rev. proxy -> strange webdav URL

### DISCUSSION THREAD

**Discussion by pongraczi** [nextcloud/all-in-one#4069] [Questions] [answered]
I run an AIO v7.9.1 (Nextcloud 27.1.5) + LDAP + rev. proxy inside an lxc container and technically it is working very well.

The strange thing is, the file access via webdav link seems not working, as expected.

Notes:   
- the desktop client is perfect, working
- editing files online (office, diagram etc.) workin as expected, 
- sharing files via public link is working

What is strange:  
- the documentation and the webdav URL under the settings of files app wrote the following link:
  - https://example.com/remote.php/dav/files/USERNAME
     - this URL is working with windows, but **not** with CAJA, NEMO etc.
     - caja error message something like this (translated back to En): `url not found. Check the url and try again.`
  -  https://example.com/remote.php/dav/webdav  (technically: davs://example.com/remote.php/dav/webdav )
    -  this one is working with CAJA, nemo, even firefox (firefox of course cannot show the webdav, but a message, which indicates it is working, but needs a valid webdav client)

I tried to find answer on the github and here, but I did not find general answer.  
I found [this post on Reddit](https://www.reddit.com/r/NextCloud/comments/10ocux8/cannot_connect_via_webdav/), but it is just about the url in case of ldap user backend, when the username is blablabla, due to ldap id.  

So, I am confused, what happening, why do I have working https://example.com/remote.php/dav/webdav and non-working https://example.com/remote.php/dav/files/USERNAME ?  

Any hints are welcome :)

**Comment by pongraczi** [comment]
Ok, just realized, there are users from LDAP and there are users from local DB.  

### Theory
Seems when the LDAP auth is enabled and is working, the non-ldap user cannot use his username anymore, because the webdav url: remote.php/dav/files/localusername expect the ldap id, instead of the username.  

Example: 
- the user name in ldap: booboo    , the uid: fadsjhlk-fkh-fsa-erreaffa (not exactly, but this is not a problem)  
  - the url:  remote.php/dav/files/booboo is invalid
  - the url:  remote.php/dav/files/fadsjhlk-fkh-fsa-erreaffa is valid, working
- the user name in DB: [chaka](https://y.yarn.co/5fad4ed0-5893-4e01-85fc-96d75255c26b_screenshot.jpg)    , the uid: does not exist
  - the url:  remote.php/dav/files/chaka is invalid
  - the url:  remote.php/dav/files/?????    is not valid, because there is no ldap uid?
  - the url:  remote.php/webdav is working

Probably this is only a side effect of using ldap.  
Seems reasonable for me.

Could someone confirm this?

**Comment by szaimen** [comment, marked as answer]
Hi, this seems to be a general Nextcloud question and not directly AIO related so please ask this at https://help.nextcloud.com. Thanks!

**Reply by pongraczi** [reply]
Thank you, you are right.
There are similar LDAP/Webdav related topics on that forum.  
After I investigated my situtation, it is definitely LDAP users/groups related and pretty normal situation, nothing special here.  

And it is recommended to not mix ldap/non-ldap users by design.
